src/librustc_trans/back/msvc/mod.rs

```diff
@@ -191,7 +214,7 @@ mod platform {
                 .map(|dir| dir.path()).collect();
             dirs.sort();
             dirs.into_iter().rev().filter(|dir| {
-                dir.join("um").is_dir()
+                dir.join("um").join("x64").join("kernel32.lib").is_file()
```

Comment:
This seems somewhat flaky (e.g. why check specifically for x64?) Could you add some comments?